src/graph_generator.cc

```diff
@@ -499,60 +569,5 @@ namespace detail {
 		}
 	}
 
-	void graph_generator::generate_horizon(task_id tid) {
-		detail::command_id lowest_prev_hid = 0;
-		for(node_id node = 0; node < num_nodes; ++node) {
-			// TODO this could be optimized to something like cdag.apply_horizon(node_id, horizon_cmd) with much fewer internal operations
```

Comment:
We lost this todo in the scattered copy-pasting of this functionality, but without actually doing it.

Reply:
This TODO was yours initially, but I don't understand what the optimization potential is in moving this to the cdag.